### PR TITLE
:bug: Source Bing Ads: campaigns, accounts, search query report streams: update schemas + date-time convert `rfc3339`

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What

Based on datadog dashboard:
- `validation_error:_.linkedagencies:_object_found_but_null_string_is_required`
- `validation_error:_.biddingscheme.maxcpc.amount:_number_found_but_null_string_is_required` 
- ` validation_error:_.costperconversion:_string_found_but_null_integer_is_required`

Convert `date` and `date-time` fields to standard `RFC3339`.
save stream_state in "native" field format

## How

- Accounts: LinkedAgencies - [customer info doc]( https://learn.microsoft.com/en-us/advertising/customer-management-service/customerinfo?view=bingads-13 ), [response soap doc](https://learn.microsoft.com/en-us/advertising/customer-management-service/searchaccounts?view=bingads-13#response-soap )
- Campaigns: Amount: [docs](https://learn.microsoft.com/en-us/advertising/campaign-management-service/maxconversionsbiddingscheme?view=bingads-13), [amount](https://learn.microsoft.com/en-us/advertising/campaign-management-service/bid?view=bingads-13&viewFallbackFrom=bingads-13%20-)
- costperconversion - changed to number as all other costPer... columns

## Recommended reading order
1. `schemas/*.json`
2. `streams.py`
3. `report_streams.py`
4. `utils.py`

## 🚨 User Impact 🚨

This version update affects next streams:

- Accounts,
- Campaigns
- Search Query Performance Report 
- AppInstallAds
- AppInstallAdLabels
- Labels
- Campaign Labels
- Keyword Labels
- Ad Group Labels
- Keywords
- Budget Summary Report
- All Hourly Reports ( end with report_hourly)

Version 2.0.0 updates schemas for Accounts, Campaigns and Search Query Performance Report streams. LinkedAgencies was changed from string to object in Accounts stream. 
BiddingScheme.MaxCpc.Amount was changed from string to number in Campaigns stream. And CostPerConversion was changed from integer to number. 
All `date` and `date-time` fields will be converted to standard `RFC3339`.

For the changes to take effect, please refresh the source schema and reset affected streams after you have applied the upgrade.

| Stream field                | Current Airbyte Type | New Airbyte Type |
|-----------------------------|----------------------|------------------|
| LinkedAgencies              | string               | object           |
| BiddingScheme.MaxCpc.Amount | string               | number           |
| CostPerConversion           | integer              | number           |

| Affected streams                                                                                                     | Field_name      | Old type                  | New type (`RFC3339`)            |
|----------------------------------------------------------------------------------------------------------------------|-----------------|---------------------------|---------------------------------|
| `AppInstallAds`, `AppInstallAdLabels`, `Labels`, `Campaign Labels`, `Keyword Labels`, `Ad Group Labels`, `Keywords`  | `Modified Time` | `04/27/2023 18:00:14.970` | `2023-04-27T16:00:14.970+00:00` |
| `Budget Summary Report`                                                                                              | `Date`          | `6/10/2021`               | `2021-06-10`                    |
| `* Report Hourly`                                                                                                    | `TimePeriod`    | `2023-11-04\|11`          | `2023-11-04T11:00:00+00:00`     |

### CODE REVIEW

**Review by davydov-d — APPROVED** [review]

**Review by artem1205 — APPROVED** [review]

**Review by alafanechere — CHANGES_REQUESTED** [review]
The change looks good to me but I believe the CAT failure is related to an expected_records failure. Shall we udpate these expected records following the schema change? 
@davydov-d told me that the `file size 260506229 exceeds limit 134217728` we see in the CAT report is often related to expected records test failure.

File: airbyte-integrations/connectors/source-bing-ads/metadata.yaml [hunk 1]
```diff
@@ -39,6 +39,13 @@ data:
           primary keys. A data reset and schema refresh of all the affected streams
           is required for the changes to take effect.
         upgradeDeadline: "2023-10-25"
+      2.0.0:
+        message:
+          Version 2.0.0 updates schemes for Accounts, Campaigns and Search Query Performance Report streams.
```
Comment (katmarkham) [inline airbyte-integrations/connectors/source-bing-ads/metadata.yaml:44]: Version 2.0.0 updates schemas for Accounts, Campaigns and Search Query Performance Report streams.
          LinkedAgencies was changed from string to object in Accounts stream. BiddingScheme.MaxCpc.Amount
          was changed from string to number in Campaigns stream. And CostPerConversion was changed from integer
          to number. Users will need to refresh the source schema and reset affected streams after upgrading.

File: docs/integrations/sources/bing-ads-migrations.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
 # Bing Ads Migration Guide
 
+## Upgrading to 2.0.0
+
+This version update only affects the Accounts, Campaigns and Search Query Performance Report streams. 
+
+Version 2.0.0 updates schemes for Accounts, Campaigns and Search Query Performance Report streams. LinkedAgencies was changed from string to object in Accounts stream.
```
Comment (katmarkham) [inline docs/integrations/sources/bing-ads-migrations.md:7]: please add chart as discusssed

**Review by katmarkham — CHANGES_REQUESTED** [review]

**Comment by darynaishchenko** [comment]
> The change looks good to me but I believe the CAT failure is related to an expected_records failure. Shall we udpate these expected records following the schema change? @davydov-d told me that the `file size 260506229 exceeds limit 134217728` we see in the CAT report is often related to expected records test failure.

@alafanechere Hi, I've merged master into my branch, tests were successful. The fields that I change in the PR are null in our expected records, so we don't need to update expected records.

**Review by alafanechere — APPROVED** [review]
Tests are passing now 👍

**Review by katmarkham — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-bing-ads/metadata.yaml [hunk 3]
```diff
@@ -39,6 +39,15 @@ data:
           primary keys. A data reset and schema refresh of all the affected streams
           is required for the changes to take effect.
         upgradeDeadline: "2023-10-25"
+      2.0.0:
+        message:
+          Version 2.0.0 updates schemas for Accounts, Campaigns, Search Query Performance Report, AppInstallAds, AppInstallAdLabels, Labels, Campaign Labels, Keyword Labels, Ad Group Labels, Keywords, Budget Summary Report and All Hourly Reports (end with report_hourly).
```
[the remark this answers is not in the document]
Comment (artem1205) [reply airbyte-integrations/connectors/source-bing-ads/metadata.yaml]: all docs updated

File: airbyte-integrations/connectors/source-bing-ads/metadata.yaml [hunk 4]
```diff
@@ -39,6 +39,11 @@ data:
           primary keys. A data reset and schema refresh of all the affected streams
           is required for the changes to take effect.
         upgradeDeadline: "2023-10-25"
+      2.0.0:
+        message:
+          Version 2.0.0 updates schemas for all hourly reports (end in report_hourly), Accounts, Campaigns, Search Query Performance Report, AppInstallAds, AppInstallAdLabels, Labels, Campaign Labels, Keyword Labels, Ad Group Labels, Keywords, and Budget Summary Report.
```
Comment (katmarkham) [inline airbyte-integrations/connectors/source-bing-ads/metadata.yaml:44]: (end in report_hourly), and the following streams: Accounts,

**Review by katmarkham — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-bing-ads/metadata.yaml (hunk 4 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-bing-ads/metadata.yaml]: updated

**Review by katmarkham — APPROVED** [review]

**Review by lazebnyi — APPROVED** [review]
lgtm

**Pull request merged**